File: Travelo.Model/Reservation.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Travelo.Model
{
    public partial class Reservation
    {
        public int Id { get; set; }
        public DateTime TimeOfReservation { get; set; }
        public int NumberOfAdults { get; set; }
        public int NumberOfChildren { get; set; }
        public int UserId { get; set; }
        public int Price { get; set; }
        public int TripItemId { get; set; }
        public int TripId { get; set; }
        public string AgencyName { get; set; }
        public string DestinationName { get; set; }
        public string CountryName { get; set; }
        public int Rating { get; set; } = 5;
        public byte[] DestinationImage { get; set; }
        public string Date { get; set; }
        public DateTime CheckIn { get; set; }
        public DateTime CheckOut { get; set; }
    }
}

File: travelo_mobile/lib/model/reservation.dart
import 'package:json_annotation/json_annotation.dart';

part 'reservation.g.dart';

@JsonSerializable()
class Reservation {
  int? id;
  DateTime? timeOfReservation;
  int? numberOfAdults;
  int? numberOfChildren;
  int? userId;
  int? price;
  int? tripItemId;
  int? tripId;
  String? agencyName;
  String? destinationName;
  String? countryName;
  int? rating;
  String? destinationImage;
  String? date;
  DateTime? checkIn;
  DateTime? checkOut;

  Reservation();

  factory Reservation.fromJson(Map<String, dynamic> json) =>
      _$ReservationFromJson(json);

  Map toJson() => {
        // 'id': id,
        // 'name': name,
        // 'countryName': countryName,
        // 'tags': tags,
        // 'image': image,
        // 'numberOfTrips': numberOfTrips,
        // 'lowestTripPrice': lowestTripPrice,
      };
}
